src/uct/ib/ud/base/ud_ep.c

```diff
@@ -684,6 +684,34 @@ static uct_ud_ep_t *uct_ud_ep_create_passive(uct_ud_iface_t *iface, uct_ud_ctl_h
     return ep;
 }
 
+static void uct_ud_ep_rx_ctl_drop_packet(uct_ud_ep_t *ep, uct_ud_neth_t *neth)
+{
+    uct_ud_iface_t *iface = ucs_derived_of(ep->super.super.iface,
+                                           uct_ud_iface_t);
+    uct_ud_ctl_hdr_t *ctl = (uct_ud_ctl_hdr_t*)(neth + 1);
+
+    ucs_assert((ctl->type == UCT_UD_PACKET_CREQ) ||
+               (ctl->type == UCT_UD_PACKET_CREP));
+    ucs_trace_data("drop %s packet (neth_psn=%u) on iface=%p ep=%p conn_sn=%d"
+                   " ep_id=%d, dest_ep_id=%d rx_psn=%u ep_flags=0x%x"
+                   " ctl_ops=0x%x rx_creq_count=%d",
```

Comment:
IMO no need to print the packet content, since it's already printed by log level data before
just "ep %p: drop duplicate CREQ, head_sn=%u"

Reply:
done

Reply:
you changed assert, but should change the trace

Reply:
but have the packet content is printed by log level data?

Reply:
log_level=data was chosen to print it only if UCX_LOG_LEVEL=data is set

Reply:
thanks for the explanation!
I forgot that we do logging of RX